worlds/terraria/__init__.py

```diff
@@ -0,0 +1,329 @@
+                for boss in mech_bosses:
+                    if state.has(boss, self.player):
+                        boss_count += 1
+                        if boss_count >= arg:
+                            return sign
+
+                return not sign
+            elif condition == "minions":
+                if type(arg) is not int:
+                    raise Exception("@minions requires an integer argument")
+
+                minion_count = 1
+                for armor, minions in armor_minions.items():
+                    if state.has(armor, self.player) and minions + 1 > minion_count:
+                        minion_count = minions + 1
+                        if minion_count >= arg:
+                            return sign
+
+                for accessory, minions in accessory_minions.items():
+                    if state.has(accessory, self.player):
+                        minion_count += minions
+                        if minion_count >= arg:
+                            return sign
+
+                return not sign
+            else:
+                raise Exception(f"Unknown function {condition}")
+        elif ty == COND_GROUP:
+            operator, conditions = condition
+            return sign == self.check_conditions(state, operator, conditions)
+
+    def check_conditions(
+        self,
+        state,
+        operator,  #: bool | None,
+        conditions,  #: list[tuple[bool, int, str | tuple[bool | None, list], str | int | None]],
+    ) -> bool:
+        if operator is None:
+            if len(conditions) == 0:
+                return True
+            if len(conditions) > 1:
+                raise Exception("Found multiple conditions without an operator")
+            return self.check_condition(state, *conditions[0])
+        elif operator:
+            return any(
+                self.check_condition(state, *condition) for condition in conditions
+            )
+        else:
+            return all(
+                self.check_condition(state, *condition) for condition in conditions
+            )
+
+    def set_rules(self) -> None:
+        for location in self.ter_locations:
+
+            def check(state: CollectionState, location=location):
+                _, _, operator, conditions = rules[rule_indices[location]]
+                return self.check_conditions(state, operator, conditions)
+
+            self.multiworld.get_location(location, self.player).access_rule = check
+
+        self.multiworld.completion_condition[self.player] = lambda state: state.has_all(
+            self.goal_items, self.player
+        )
+
+    def generate_basic(self) -> None:
+        for location, item in self.locked_items.items():
+            self.multiworld.get_location(location, self.player).place_locked_item(item)
+
+    def fill_slot_data(self):  # -> dict[str, object]:
+        return {
+            "goal": list(self.goal_locations),
+            "deathlink": bool(self.multiworld.deathlink[self.player]),
```

Comment:
Follow to my comment in Options.py, you can leave it `deathlink` in your slot data so you don't need to update your client, but would change `self.multiworld.deathlink` to `self.multiworld.death_link` if you make the requested change.

```suggestion
            "deathlink": bool(self.multiworld.death_link[self.player]),
```